src/plugins/intellisense/view.tsx

```diff
@@ -0,0 +1,486 @@
+                .partialFunctionCallIdent()
+                ?.params.map((e, i) => [e, i] as const) ?? []
+            }
+            key={(e) => e[0]}
+          >
+            <div class="pfc-params">
+              {(p: [string, number]) => {
+                return (
+                  <div
+                    class={() =>
+                      this.props.partialFunctionCall()?.paramIndex === p[1]
+                        ? "pfc-param-selected"
+                        : "pfc-param"
+                    }
+                  >
+                    <DStaticMathquillView
+                      config={{}}
+                      latex={() =>
+                        addBracketsToIdent(p[0]) +
+                        (p[1] ===
+                        (this.props.partialFunctionCallIdent()?.params
+                          ?.length ?? 0) -
+                          1
+                          ? ""
+                          : ",")
+                      }
+                    ></DStaticMathquillView>
+                  </div>
+                );
+              }}
+            </div>
+          </For>
+          <StaticMathQuillView latex=")"></StaticMathQuillView>
+        </div>
+      </div>
+    );
+  }
+}
+
+let counter2 = 0;
+
+export class View extends Component<{
+  x: () => number;
+  y: () => number;
+  idents: () => { idents: BoundIdentifier[] }[];
+  autocomplete: (option: BoundIdentifier) => void;
+  index: () => number;
+  partialFunctionCall: () => PartialFunctionCall | undefined;
+  partialFunctionCallIdent: () => BoundIdentifierFunction | undefined;
+  partialFunctionCallDoc: () => string | undefined;
+  show: () => boolean;
+  jumpToDefinition: (str: string) => void;
+}> {
+  init() {}
+
+  template() {
+    return (
+      <div
+        id="intellisense-container"
+        class={() =>
+          this.props.index() >= 0 ? "selected-intellisense-container" : ""
+        }
+        style={() => ({
+          top: (this.props.y() + 30).toString() + "px",
+          left: this.props.x().toString() + "px",
+          display:
+            (this.props.idents().length > 0 ||
+              this.props.partialFunctionCall()) &&
+            this.props.show()
+              ? "block"
+              : "none",
+          transform:
+            this.props.y() > window.innerHeight / 2
+              ? `translateY(calc(-100% - 50px))`
+              : "",
+        })}
+        onClick={(e: MouseEvent) => {
+          e.stopPropagation();
+        }}
+      >
+        <PartialFunctionCallView
+          partialFunctionCall={() => this.props.partialFunctionCall()}
+          partialFunctionCallIdent={() => this.props.partialFunctionCallIdent()}
+          partialFunctionCallDoc={() => this.props.partialFunctionCallDoc()}
+        ></PartialFunctionCallView>
+        <For
+          each={() =>
+            this.props.idents().map((ident, index) => ({ ...ident, index }))
+          }
+          key={() => counter2++}
```

Comment:
`counter2++` sus. Would ``key={() => `${ident}:${index}`} `` work?

Reply:
Update: Tried using the identifier name as a key; it didn't work. Not totally sure why this is. From what I recall testing this earlier, I suspect it's the fact that I need to keep track of index information for each element of the array (so I know which component to highlight). Because DCGView doesn't re-run the function for rendering the "list element" component if it already has its index, subsequent re-renders use an outdated array index, which as you can imagine really messes things up.

Reply:
I could probably fix this by making the array index reactive by wrapping it in a function. I might make a custom `IndexFor` component of sorts for dealing with this sort of behavior.

Reply:
Update: `IndexFor` seems to have worked!